### PR TITLE
Rwc2022

### PR DESCRIPTION
**Pull request by PetervDooren**

### CODE REVIEW

File: launch/hero_bringup.launch [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <arg name="robot_name" default="hsrb" />
     <arg name="personal_name" default="hero" />
     <arg name="enable_emergency_stop" default="false" />
+    <arg name="gui" default="false" />  <!-- Gazebo GUI -->
```
Comment (MatthijsBurgh) [inline launch/hero_bringup.launch:24]: Call it `gazebo_gui` as just `gui` is not clear

File: launch/start.launch [hunk 2]
```diff
@@ -24,6 +24,7 @@
     <arg name="robot_name" default="hsrb" />
     <arg name="personal_name" default="hero" />
     <arg name="enable_emergency_stop" default="false" />
+    <arg name="gui" default="false" />  <!-- Gazebo GUI -->
```
Comment (MatthijsBurgh) [inline launch/start.launch:27]: Change to `gazebo_gui` and add comment it is not forwarded to the hardware manager.

File: launch/state_machines/challenge_carry_my_luggage.launch [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0"?>
+<launch>
+    <arg name="robot_name" value="hero" />
+
+    <remap from="/diagnostics" to="/$(arg robot_name)/diagnostics" />
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_carry_my_luggage.launch:5]: Why is this needed? This is undesired.

File: launch/state_machines/challenge_clean_the_table.launch [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<launch>
+    <arg name="robot_name" value="hero" />
+
+    <remap from="/diagnostics" to="/$(arg robot_name)/diagnostics" />
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_clean_the_table.launch:5]: Why is this needed? This is undesired.

File: launch/state_machines/challenge_clean_the_table.launch [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+        <!-- PEOPLE RECOGNITION -->
+        <include file="$(find robot_launch_files)/launch/world_modeling/people_recognition.launch">
+            <arg name="robot_name" value="$(arg robot_name)" />
+        </include>
+    </group>
+
+    <!-- Start the challenge -->
+    <node pkg="challenge_clean_the_table" type="challenge_clean_the_table" name="state_machine" args="hero" output="screen" required="true" />
+</launch>
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_clean_the_table.launch:28]:
```suggestion
    <!-- Start the challenge -->
    <node pkg="challenge_clean_the_table" type="challenge_clean_the_table" name="state_machine" args="hero" output="screen" required="true" />

</launch>
```

File: launch/state_machines/challenge_final.launch [hunk 6]
```diff
@@ -24,11 +24,9 @@
...
         <include file="$(find robot_launch_files)/launch/interaction/telegram.launch">
-             <arg name="robot_name" value="$(arg robot_name)" />
-             <arg name="caption_as_frame_id" value="true" />
+            <arg name="robot_name" value="$(arg robot_name)" />
+            <arg name="caption_as_frame_id" value="true" />
         </include>
 
     </group>
-
-    <!-- Don't start the challenge, launch in different terminal so that we can verify the deps are working before challenge -->
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_final.launch:33]: Why remove it, when the actual challenge node is not added?

File: launch/state_machines/challenge_hand_me_that.launch [hunk 7]
```diff
@@ -25,4 +25,5 @@
     </group>
 
     <!-- Don't start the challenge, launch in different terminal so that we can verify the deps are working before challenge -->
+    <node pkg="challenge_hand_me_that" type="challenge_hand_me_that" name="state_machine" args="$(arg robot_name)" output="screen" required="true" />
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_hand_me_that.launch:28]: This is conflicting

Comment (MatthijsBurgh) [reply launch/state_machines/challenge_hand_me_that.launch:28]: When you keep the challenge node, add empty line between node and closing launch tag.

File: launch/state_machines/challenge_serve_breakfast.launch [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+        <!-- PEOPLE RECOGNITION -->
+        <include file="$(find robot_launch_files)/launch/world_modeling/people_recognition.launch">
+            <arg name="robot_name" value="$(arg robot_name)" />
+        </include>
+    </group>
+
+    <!-- Start the challenge -->
+    <node pkg="challenge_serve_breakfast" type="challenge_serve_breakfast" name="state_machine" args="hero" output="screen" required="true" />
+</launch>
```
Comment (MatthijsBurgh) [inline launch/state_machines/challenge_serve_breakfast.launch:28]:
```suggestion
    <node pkg="challenge_serve_breakfast" type="challenge_serve_breakfast" name="state_machine" args="hero" output="screen" required="true" />

</launch>
```

File: package.xml [hunk 9]
```diff
@@ -14,6 +14,7 @@
 
   <buildtool_depend>catkin</buildtool_depend>
 
+  <exec_depend>depthimage_to_laserscan</exec_depend>
```
Comment (MatthijsBurgh) [inline package.xml:17]: This is a launch dependency, not a software dependency, so should not be in the `package.xml`

File: parameters/tools/rviz_config.rviz [hunk 10]
```diff
@@ -625,29 +625,38 @@ Visualization Manager:
...
-      Value: false
+      Value: true
+    - Class: rviz/MarkerArray
+      Enabled: true
+      Marker Topic: /markers
+      Name: MarkerArray
+      Namespaces:
+        "": true
+      Queue Size: 100
+      Value: true
```
Comment (MatthijsBurgh) [inline parameters/tools/rviz_config.rviz:659]: What markers?

File: parameters/tools/rviz_config.rviz [hunk 11]
```diff
@@ -625,29 +625,38 @@ Visualization Manager:
...
       Topic: /hero/ssl/direction_of_arrival
       Unreliable: false
       Value: true
-    - Alpha: 0.699999988
+    - Alpha: 0.699999988079071
       Class: rviz/Map
       Color Scheme: map
       Draw Behind: false
-      Enabled: false
+      Enabled: true
```
Comment (MatthijsBurgh) [inline parameters/tools/rviz_config.rviz:646]: I think Gmapping map should be disabled by default

File: parameters/tools/rviz_config.rviz [hunk 12]
```diff
@@ -625,29 +625,38 @@ Visualization Manager:
       Value: true
     - Alpha: 1
       Axes Length: 1
-      Axes Radius: 0.100000001
+      Axes Radius: 0.10000000149011612
```
Comment (MatthijsBurgh) [inline parameters/tools/rviz_config.rviz:628]: We should merge these small changes.

File: parameters/tools/rviz_config.rviz (hunk 10 above)
Comment (MatthijsBurgh) [reply parameters/tools/rviz_config.rviz:659]: All changes in this file can be reverted by changing 02e7c5d9a12d495f3c6ec99ebdecb499e956d15d

File: parameters/world_modeling/world_model_plugin_laser_clearer.yaml [hunk 13]
```diff
@@ -2,4 +2,4 @@ name: entity_clearer
 type: ed_sensor_integration/clearer
 enabled: 1
 parameters:
-    entity_timeout: 1.0
+    entity_timeout: 5
```
Comment (MatthijsBurgh) [inline parameters/world_modeling/world_model_plugin_laser_clearer.yaml:5]: Do we want to keep this @ar13pit @LarsJanssenTUe

File: scripts/record_video_from_camera.py [hunk 14]
```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/env python3
```
Comment (MatthijsBurgh) [inline scripts/record_video_from_camera.py:1]: This is too generic to keep here

File: scripts/services/hero2/hero2-rgbd-shm-server.service [hunk 15]
```diff
@@ -6,7 +6,7 @@ WantedBy=multi-user.target
 
 [Service]
 User=amigo
-Restart=on-failure
+Restart=always
```
Comment (MatthijsBurgh) [inline scripts/services/hero2/hero2-rgbd-shm-server.service:9]: @MatthijsBurgh @ar13pit discuss this change

File: scripts/services/hero1/hero1-microphone-settings.service [hunk 16]
```diff
@@ -0,0 +1,12 @@
+[Unit]
```
Comment (MatthijsBurgh) [inline scripts/services/hero1/hero1-microphone-settings.service:1]: @MatthijsBurgh  @ar13pit discuss this

File: scripts/services/hero1-demo/hero1-demo-microphone-settings.service [hunk 17]
```diff
@@ -0,0 +1,12 @@
+[Unit]
```
Comment (MatthijsBurgh) [inline scripts/services/hero1-demo/hero1-demo-microphone-settings.service:1]: Same as non-demo file

File: launch/hero_bringup.launch [hunk 18]
```diff
@@ -109,6 +110,7 @@
             <arg name="robot_name" value="$(arg robot_name)" />
             <arg name="personal_name" value="$(arg personal_name)" />
             <arg name="param_file" value="$(arg param_file)" />
+            <arg name="gui" value="$(arg gui)" />
```
Comment (MatthijsBurgh) [inline launch/hero_bringup.launch:113]: Also change to `gazebo_gui`

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: launch/state_machines/challenge_carry_my_luggage.launch (hunk 3 above)
Comment (PetervDooren) [reply launch/state_machines/challenge_carry_my_luggage.launch:5]: tradition. It is present in the other challenge files as well.

File: launch/hero_bringup.launch (hunk 1 above)
Comment (PetervDooren) [reply launch/hero_bringup.launch]: done

File: launch/start.launch (hunk 2 above)
Comment (PetervDooren) [reply launch/start.launch]: done

File: scripts/services/hero2/hero2-rgbd-shm-server.service (hunk 15 above)
Comment (PetervDooren) [reply scripts/services/hero2/hero2-rgbd-shm-server.service:9]: We should keep this. There should not be a reason for the shm server to exit cleanly as you want it to run continuously

File: scripts/services/hero1/hero1-microphone-settings.service (hunk 16 above)
Comment (PetervDooren) [reply scripts/services/hero1/hero1-microphone-settings.service:1]: Do we need to keep this if we transition to picovoice? If not let's drop these

File: launch/state_machines/challenge_carry_my_luggage.launch (hunk 3 above)
Comment (MatthijsBurgh) [reply launch/state_machines/challenge_carry_my_luggage.launch:5]: Ok, I see it is indeed there and needed. https://github.com/tue-robotics/hero_bringup/commit/bfaa745f07b7d36829db4718f7b546656f0fa2ef

**Comment by PetervDooren** [comment]
@MatthijsBurgh @ar13pit, could you have a look at the review comments?

File: scripts/services/hero1-demo/hero1-demo-microphone-settings.service (hunk 17 above)
Comment (MatthijsBurgh) [reply scripts/services/hero1-demo/hero1-demo-microphone-settings.service:1]: Keeping it for now. Will be removed when migrating to picovoice.

File: scripts/services/hero1/hero1-microphone-settings.service (hunk 16 above)
Comment (MatthijsBurgh) [reply scripts/services/hero1/hero1-microphone-settings.service:1]: Keeping it for now. Will be removed when migrating to picovoice.

File: scripts/services/hero2/hero2-rgbd-shm-server.service (hunk 15 above)
Comment (MatthijsBurgh) [reply scripts/services/hero2/hero2-rgbd-shm-server.service:9]: Yes, we should keep this. Especially with the pending RGBD changes, this is required.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
We still need to lay an egg about the navigation parameters and the laser clearer.

**Review by MatthijsBurgh — APPROVED** [review]
